src/native/libs/System.Native/pal_io.c

```diff
@@ -441,10 +445,17 @@ static const size_t dirent_alignment = 8;
 int32_t SystemNative_GetReadDirRBufferSize(void)
 {
 #if HAVE_READDIR_R
+    size_t result = sizeof(struct dirent);
```

Comment:
Maybe just use uint32_t here? (instead of size_t) since that's what this will return, and is large enough.
Note size_t is 64-bits with our compilation environment.

Not a "stopper" comment, just a suggestion.  Looks fine to me overall.

Reply:
C11 section 6.5.3.4

>    The `sizeof` operator yields the size (in bytes) of its operand, which may be an expression or the parenthesized name of a type. The size is determined from the type of the operand. The result is an integer. If the type of the operand is a variable length array type, the operand is evaluated; otherwise, the operand is not evaluated and the result is an integer constant. <ins>The type of the result is `size_t`.</ins>

Reply:
> C11 section 6.5.3.4

Right, so it would still need the cast.  I'd have done it here, that's all.

Reply:
> Note size_t is 64-bits with our compilation environment.

This line of code is complied for all platforms not only for TARGET_SUNOS. Explicit cast at the end (where it is actually needed) is fine. It all compiles to a constant value on illumos.